### PR TITLE
Add search capability to recommendations section

### PR DESCRIPTION
**Pull request by alexcn3**
Screenshot 2020-06-07 at 8 22 07 PM
Screenshot 2020-06-07 at 8 21 58 PM
Screenshot 2020-06-07 at 8 21 44 PM

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -124,3 +124,14 @@ async function getSpecificSug(category) {
   const recs = await response.json();
   showRecs(recs);
 }
+
+async function getSearchResults() {
+  const params = new URLSearchParams();
+  const searchForm = document.getElementById('search-form');
+  const searchTerm = searchForm.elements[0].value;
+  console.log(searchTerm);
```
Comment (eyh8) [inline portfolio/src/main/webapp/script.js]: Get rid of this?

File: portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java [hunk 2]
```diff
@@ -35,29 +35,34 @@
...
+        showResults(response, "", userSearch);
+      }
     } else {
-       showResults(response, userCategory);
+       showResults(response, userCategory, "");
     }
   }
 
-  public void showResults(HttpServletResponse response, String sugCategory) throws IOException {
+  public void showResults(HttpServletResponse response, String sugCategory, String sugSearch) throws IOException {
```
Comment (eyh8) [inline portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java:66]: Consider using Optional if they could be empty

**Review by eyh8 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java (hunk 2 above)
Comment (alexcn3) [reply portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java:66]: I made the parameters null but I'm not sure if that's what you meant. From what I've read, the Optional class in java still doesn't allow you to pass in empty parameters but I could be looking at the wrong thing.

Comment (eyh8) [reply portfolio/src/main/java/com/google/sps/servlets/SuggestServlet.java:66]: You can have the method signature be
```
public void showResults(HttpServletResponse response, Optional<String> sugCategory, Optional<String> sugSearch)
```

And if you'd like to pass in empty Strings, you can do:

```
showResults(response, Optional.empty(), Optional.empty())
```

**Review by eyh8 — APPROVED** [review]

**Pull request merged**